src/units/abilities.cpp

```diff
@@ -1338,23 +1333,61 @@ static bool overwrite_special_affects(const config& special)
 	return (apply_to == "one_side" || apply_to == "both_sides");
 }
 
-unit_ability_list attack_type::overwrite_special_checking(unit_ability_list input, unit_ability_list overwriters, bool is_special) const
+unit_ability_list attack_type::overwrite_special_overwriter(unit_ability_list overwriters, const std::string& tag_name) const
 {
-	for(unit_ability_list::iterator i = overwriters.begin(); i != overwriters.end();) {
-		if(!overwrite_special_affects(*i->ability_cfg)) {
-			i = overwriters.erase(i);
-		} else {
-			++i;
+	std::vector<double> priorityvec;
+	int list_count = 0;
+	utils::erase_if(overwriters, [&](const unit_ability& i) {
+		if(overwrite_special_affects(*i.ability_cfg)){
+			list_count += 1;
+			auto overwrite_specials = (*i.ability_cfg).optional_child("overwrite");
+			if(overwrite_specials && !overwrite_specials["priority"].empty()){
+				priorityvec.push_back(overwrite_specials["priority"].to_double(0));
+			}
 		}
-	}
+		return (!overwrite_special_affects(*i.ability_cfg));
+	});
+
 	if(overwriters.empty()){
-		return input;
+		return overwriters;
+	}
+
+	if(!priorityvec.empty() && (list_count >= 2)){
+		if(priorityvec.size() >= 2){
+			std::sort(priorityvec.begin(), priorityvec.end(),[](const double &l, const double &r)
+				{
+					return l > r;
+				}
+			);
+		}
+		for(double priority : priorityvec){
+			if(priority > 0){
+				unit_ability_list temp_overwriters;
+				std::pair<unit_ability_list, int> overwrite_list;
+				temp_overwriters.append_if(overwriters, [&](const unit_ability& i) {
+					auto overwrite_specials = (*i.ability_cfg).optional_child("overwrite");
+					return (overwrite_specials && overwrite_specials["priority"].to_double(0) == priority);
+				});
```

Comment:
Why is this only appending to `temp_overwriters` if it matches a priority from `priorityvec`? Is there any case where there would be no matching priority?

Reply:
For each value of the vector processed, only the elements having this priority value can be used to decide which elements of lower priority will be removed or not, once this processing has been carried out the list cleared of these elements serves again as a check to constitute the temporary list made up of following priority items where they still exist. let's take the example of a1,a2 and a3 which all have overwrite_specials but a1 has priority over a2 and a3 and a2 has priority over a3 but the filters of a1 do not correspond to a3 but on the other hand a2 can overwrite a3, if I use the list complete as a reference for its own splicing a1 overwrites a2 but a2overwrites a3 in the output list because the input list does not change. with a temporary list we first have a1 which overwrites a2 but not a3, a2 cannot overwrite a3 because it is already overwritten when there since the priority value which corresponds to it no longer corresponds to the list at the end of checking by a1 , so a1 and a3 are found in the final overwriter list